pgxgeom.go

```diff
@@ -69,9 +70,123 @@ func (c codec) PreferredFormat() int16 {
 	return pgtype.BinaryFormatCode
 }
 
+// GeomScanner enables PostGIS geometry/geography values to be scanned into
+// arbitrary Go types. For more context, see section "Extending Existing
+// PostgreSQL Type Support" of the README for jackc/pgx/v5/pgtype.
+type GeomScanner interface {
```

Comment:
Good question. I think that they should be exported. Reasons: (1) it is semantically part of the public API and should be documented so that users know what methods they need to implement on their custom types; (2) this follows the pattern set by first-party codecs like [`pgtype.TimeValuer`] / [`pgtype.TimeScanner`], which are exported; and (3) in my code that declares custom types that implement these interfaces, I am declaring

```go
var _ pgxgeom.GeomValuer = &MyMultiLineString{}
var _ pgxgeom.GeomScanner = &MyMultiLineString{}
```

to check at compile time that I have implemented the signatures correctly, which also requires that these symbols be exported.

If you disagree, let me know and I'm happy to amend; I have a preference but do not feel strongly about it.

[`pgtype.TimeValuer`]: https://pkg.go.dev/github.com/jackc/pgx/v5/pgtype#TimeValuer
[`pgtype.TimeScanner`]: https://pkg.go.dev/github.com/jackc/pgx/v5/pgtype#TimeScanner